### PR TITLE
Newsletter signup sticky footer

### PR DESCRIPTION
**Pull request by Guntrisoft**
screen shot 2015-08-06 at 10 20 37
screen shot 2015-08-06 at 10 19 37

### CODE REVIEW

File: app/assets/stylesheets/components/common/_news_signup_sticky.scss [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  position: relative;
+}
+
+.news-signup-sticky--positioned {
+  visibility: hidden;
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  width: 100%;
+  z-index: 100;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_news_signup_sticky.scss]: Worth declaring the z-index layers as variables?

File: app/views/shared/_news_signup_sticky.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div class="news-signup-sticky<% if local_assigns.has_key?(:fixed_position) %> news-signup-sticky--positioned<% end %>">
+    <p class="news-signup-sticky__intro">200,000 people are taking care of their money with our FREE money advice newsletter.</p>
+    <p class="news-signup-sticky__intro news-signup-sticky__intro--second">Are you? Sign up today.</p>
+
+    <div class="news-signup-sticky__inner">
+        <%= form_tag(main_app.newsletter_subscription_path, class: 'js-news-signup-test-form', remote: true) do %>
```
Comment (benbarnett) [inline app/views/shared/_news_signup_sticky.html.erb]: Too much indentation here by the looks of it

File: app/assets/stylesheets/components/common/_news_signup_sticky.scss (hunk 1 above)
Comment (Guntrisoft) [inline app/assets/stylesheets/components/common/_news_signup_sticky.scss]: True - is this a 'thing' on the frontend codebase? I remember the blog had z-indexes set here: https://github.com/moneyadviceservice/publify/blob/5531b58545705e971a5b9a9c3bc765c293ecf3cc/app/assets/stylesheets/_settings.scss#L75

Comment (Guntrisoft) [inline app/assets/stylesheets/components/common/_news_signup_sticky.scss]: I notice a lot of z-index: [0-9] going on in the frontend codebase - worth a seperate PR to tidy up maybe?

Comment (benbarnett) [inline app/assets/stylesheets/components/common/_news_signup_sticky.scss]: Yeah sounds cool

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -99,6 +99,14 @@ require(['common', 'jquery'], function(MAS, $) {
         $('.cookie-message').hide();
         $('.footer-site-links__cookie-link').removeClass('is-on');
       });
+
+      $('.news-signup-sticky form').bind('ajax:error', function(e) {
```
Comment (benbarnett) [inline app/assets/javascripts/application.js]: What happens on success?

Comment (Guntrisoft) [inline app/assets/javascripts/application.js]: https://github.com/moneyadviceservice/frontend/pull/1149/files#diff-7ece0453f0fe34cd7244f25ff9008fb6R13

File: app/assets/stylesheets/components/common/_news_signup_sticky.scss [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  }
+
+  @include respond-to($mq-m) {
+    display: inline-block;
+    margin: 0 $baseline-unit*2;
+    position: relative;
+    top: 1px;
+  }
+
+  &::-webkit-input-placeholder {
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_news_signup_sticky.scss]: Can you see what autoprefixer does with this? There are other prefixes I believe.

Comment (Guntrisoft) [inline app/assets/stylesheets/components/common/_news_signup_sticky.scss]: Spot on, ::placeholder FTW

**Comment by Tr4pSt3R** [comment]
Is this being closed so that we can open a new one?

**Comment by nmiddleweek** [comment]
Yes

**Pull request closed without merging**